packages/secure_storage/amplify_secure_storage_dart/lib/src/types/macos_secure_storage_options.dart

```diff
@@ -26,29 +27,41 @@ abstract class MacOSSecureStorageOptions
         Built<MacOSSecureStorageOptions, MacOSSecureStorageOptionsBuilder> {
   /// {@macro amplify_secure_storage_dart.macos_secure_storage_options}
   ///
+  /// #### [useDataProtection]
   /// {@template amplify_secure_storage_dart.macos_secure_storage_options.useDataProtection}
-  /// [useDataProtection] sets the [`kSecUseDataProtectionKeychain`](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps)
-  /// attribute for all Keychain operations. Set this to true to
-  /// enforce that Keychain items behave like iOS Keychain items.
+  /// Sets the [`kSecUseDataProtectionKeychain`](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps)
+  /// attribute for all Keychain operations.
+  ///
+  /// This value defaults to true to enforce that
+  /// MacOS Keychain items behave like iOS Keychain items.
+  /// This prevents other applications from accessing
+  /// or modifying items created by the current application.
+  ///
+  /// **Warning:** This should not be disabled in production applications.
+  /// This value is only exposed for testing. Disabling it lowers
+  /// the security of your application and can cause secure storage
+  /// to have unexpected behaviors.
   /// {@endtemplate}
   ///
+  /// #### [accessible]
   /// {@template amplify_secure_storage_dart.macos_secure_storage_options.accessible}
-  /// [accessible] sets the [kSecAttrAccessible](https://developer.apple.com/documentation/security/ksecattraccessible)
+  /// Sets the [kSecAttrAccessible](https://developer.apple.com/documentation/security/ksecattraccessible)
   /// attribute for all Keychain operations.
   /// {@endtemplate}
   ///
+  /// #### [accessGroup]
   /// {@template amplify_secure_storage_dart.macos_secure_storage_options.accessGroup}
-  /// [accessGroup] sets the [kSecAttrAccessGroup](https://developer.apple.com/documentation/security/ksecattraccessgroup)
+  /// Sets the [kSecAttrAccessGroup](https://developer.apple.com/documentation/security/ksecattraccessgroup)
   /// attribute for all Keychain operations.
   /// {@endtemplate}
   ///
   /// **Note:** [useDataProtection] must be set to true when specifying an [accessGroup].
   ///
-  /// **Note:** When [useDataProtection] is set to true, the app must be added to one
+  /// **Note:** Unless [useDataProtection] is set to false, the app must be added to one
   /// or more Keychain Access Groups in Xcode. For more info, see the
   /// [Platform Setup docs](https://docs.amplify.aws/lib/project-setup/platform-setup/q/platform/flutter/).
   factory MacOSSecureStorageOptions({
-    bool useDataProtection = true,
+    @visibleForTesting bool useDataProtection = true,
```

Comment:
If it's visiblefortesting only should it even be documented publicly?

I've been thinking about this generally, too. What about `bool useDataProtection = zReleaseMode` which would allow easier onboarding since even testing Auth apps locally is a pain having to configure Xcode and remember not to commit those changes.

Reply:
Re: should it be documented - Maybe not. I think it would be valid for a customer to use it in a test if they wanted, but not sure we need to document it.

It would allow easier onboarding, but potentially lead to confusion down the line. I could see a customer not realizing there is a difference in release build and being pretty confused why their first release build has an issue. I have also stumbled upon some odd differences in keychain behavior when this is true/false (See [this comment](https://github.com/aws-amplify/amplify-flutter/pull/2118/files#diff-0454077cd10a395b9edaaca51c76b71f0504991655de69ed05f185dfd94aced5R178-R182)). That is why I added the `@visibleForTesting`. I want to really discourage customers from using it given the differences in keychain behavior. I actually want to update our tests to have this set to `true`. I have been running the tests locally with it set to true, but would like to get that in CI given the differences in behavior I have seen.

Reply:
Bummer, okay. That makes sense.

Reply:
I think eventually, then, we should consider removing the parameter altogether.